aimarker.js

```
@@ -0,0 +1,838 @@
+  const insertText = (tag, closeTag = '') => {
+    onChange((value || '') + `${tag}text${closeTag}`);
+  };
+
+  const renderPreview = (text) => {
+    if (!text) return <span className="text-slate-400 italic">Nothing typed yet...</span>;
+    let html = text
+      .replace(/\*\*(.*?)\*\*/g, '<b>$1</b>')
+      .replace(/\*(.*?)\*/g, '<i>$1</i>')
+      .replace(/\$(.*?)\$/g, '<span class="font-mono bg-yellow-50 text-yellow-700 px-1 rounded border border-yellow-200 text-xs">$1</span>') // Fake LaTeX
+      .replace(/\n/g, '<br/>');
+    return <div dangerouslySetInnerHTML={{ __html: html }} />;
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)